src/Lucene.Net/Search/Searcher.cs

```diff
@@ -143,15 +143,56 @@ public virtual Explanation Explain(Query query, int doc, IState state)
 	    /// <seealso cref="Lucene.Net.Search.Similarity.Default">
 	    /// </seealso>
 	    public virtual Similarity Similarity
-	    {
-	        get { return this.similarity; }
-	        set { this.similarity = value; }
-	    }
-
-	    /// <summary> creates a weight for <c>query</c></summary>
-		/// <returns> new weight
-		/// </returns>
-		public /*protected internal*/ virtual Weight CreateWeight(Query query, IState state)
+        {
+            get
+            {
+                if ( usingLightWeightSimilarity)
+                {
+                    return LightWeightSimilarity.Instance;
+                }
+
+                return similarity;
+            }
+            set => similarity = value;
+        }
+
+        /// <summary>
+        /// ThreadStatic flag to indicate if we want to disable idfExplanation
+        /// </summary>
+        [ThreadStatic]
+        protected static bool usingLightWeightSimilarity;
+
+        /// <summary>
+        /// Enable the light weight similarity feature
+        /// </summary>
+        /// <returns>A disposable object that disable the light weight feature on dispose</returns>
+        public static IDisposable EnableLightWeightSimilarity()
+        {
+            return new LightWeightSimilarityScope();
+        }
+
+        /// <summary>
+        /// A thread local scope that will enable us to use a lighter version of the similarity object
+        /// </summary>
+        private class LightWeightSimilarityScope : IDisposable
```

Comment:
Make this a `struct`

Reply:
But I couldn't use parameterless constructor it looks... ugly